File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.js
function solution(n, m, section) {
    const arr = new Array(n).fill(1);
    
    for (let i of section) {
        arr[i-1] = 0;
    }
    
    let result = 0;
    for (let i = 0; i < n; i++) {
        if (arr[i] === 0) {
            result++;
            for (let j = 0; j < m; j++) {
                if (i + j > n - 1) {
                    break;
                }
                arr[i+j] = 1;
            }
        }
    }
    return result;
}

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.py
import math

def solution(n, m, section):
    # result = section[-1] - section[0] + 1
    # return math.ceil(result / m)
    arr = list(1 for i in range(n))
    for i in section:
        arr[i-1] = 0
    
    result = 0
    for i in range(n):
        if arr[i] == 0:
            result += 1
            for j in range(m):
                if i + j > n - 1:
                    break;
                arr[i+j] = 1
    return result
    
        
    